rubocop/cop/i18n/static_translation_definition.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module I18n
+      # This cop flags translation definitions in static scopes because changing
+      # locales has no effect and won't translate this text again. The rule here is based upon
+      # https://gitlab.com/gitlab-org/gitlab/-/blob/master/rubocop/cop/static_translation_definition.rb?ref_type=heads.
+      #
+      # @example
+      #   # bad
+      #   class MyExample
```

Comment:
To go with the point about `t` being preferred over `I18n.t` in controllers and views, maybe this class should be `ExampleController` so that `t` would actually work here.

Generally this example class is not great because it uses methods in the good example that are never used by us and have better alternatives. For example, see the comment on `validates` and consider that we usually do extra error messages via something like `self.errors.add(:base, :blocked_gifts, byline: gift.pseud.byline)`, which (like the validate example) doesn't call `t` at all but instead works with Rails magic to look up translations via a symbol.

Are translations in static scopes an issue for us at all? If not, it may be better to just not add this cop because it uses so much code in the good example that I would consider quite questionable.

Or maybe these examples could show only the `validates` case, since that is what the rspec test checks for. The rest of the checks could stay, but I would not include them in the example, especially not the "good" case.

Reply:
Looking at this example class again - it would be great to use `"string"` instead of `'string'`, so that rubocop won't immediately complain about that when someone copies the example.